### PR TITLE
use `coreth/rpc`

### PR DESCRIPTION
**Pull request by dhrubabasu**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -45,17 +45,14 @@ type client struct {
...
-	eth, err := NewEthClient(endpoint)
-	if err != nil {
-		return nil, err
-	}
+	eth, err := NewEthClient(ctx, endpoint)
 
 	return client{
 		Client:         info.NewClient(endpoint),
 		EthClient:      eth,
 		ContractClient: NewContractClient(eth.Client),
```
Comment (patrick-ogrady) [inline client/client.go:56]: This will panic as `eth` could be `nil`

File: client/eth.go [hunk 2]
```diff
@@ -18,52 +18,41 @@ var (
...
-	c, err := ethclient.Dial(endpointURL)
-	if err != nil {
-		return nil, err
-	}
+	c, err := rpc.DialContext(ctx, endpointURL)
 
 	return &EthClient{
-		Client: c,
-		rpc:    rpc.NewRPCRequester(endpoint),
+		Client: ethclient.NewClient(c),
```
Comment (patrick-ogrady) [inline client/eth.go:32]: Could panic if `c` is nil...just return err

Comment (dhrubabasu) [reply client/eth.go:32]: This wouldn't panic - reverting these changes so it's more obvious

**Review by patrick-ogrady — APPROVED** [review]
Tested on mainnet 👍

**Pull request merged**